crates/project_panel/src/project_panel.rs

```diff
@@ -1274,6 +1293,13 @@ impl ProjectPanel {
                     })??
                     .await?;
                 }
+                if let Some(next_selection) = next_selection {
+                    this.update(&mut cx, |this, cx| {
+                        this.selection = Some(next_selection);
+                        this.autoscroll(cx);
+                        cx.notify();
```

Comment:
```suggestion
```

That seems to be done in the `autoscroll`.

Optional: I personally find `this` less readable than the real name of the variable, so we can change it along the way if you think so too.